src/integrations/terminal/ExecaTerminalProcess.ts

```diff
@@ -45,18 +46,48 @@ export class ExecaTerminalProcess extends BaseTerminalProcess {
 			this.terminal.setActiveStream(stream, subprocess.pid)
 
 			for await (const line of stream) {
+				if (this.aborted) {
+					break
+				}
+
 				this.fullOutput += line
 
 				const now = Date.now()
 
-				if (this.isListening && (now - this.lastEmitTime_ms > 250 || this.lastEmitTime_ms === 0)) {
+				if (this.isListening && (now - this.lastEmitTime_ms > 500 || this.lastEmitTime_ms === 0)) {
 					this.emitRemainingBufferIfListening()
 					this.lastEmitTime_ms = now
 				}
 
 				this.startHotTimer(line)
 			}
 
+			if (this.aborted) {
```

Comment:
We now stop processing the terminal output when an abort signal is received. If the abort signal can't kill the process then we manually kill it.